client/state/data-layer/wpcom-api-middleware.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * External dependencies
+ */
+import { has } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import handlers from './wpcom';
+
+export const middleware = store => next => action =>
+	has( handlers, action.type )
+		? handlers[ action.type ].forEach( handler => handler( store, action ) )
```

Comment:
I have updated this in 4e48488 with the softer policy of allowing non-array values. I feel dirtier having done so, as it bothers me to mix types. nevertheless, it's here for feedback. before we merge my inclination is to change it back to _only_ allowing arrays of functions.